sqlglot/expressions.py

```diff
@@ -82,15 +82,15 @@ class Expression(metaclass=_Expression):
 
     key = "expression"
     arg_types = {"this": True}
-    __slots__ = ("args", "parent", "arg_key", "comments", "_type", "_meta")
+    __slots__ = ("args", "parent", "arg_key", "comments", "_type", "meta")
 
     def __init__(self, **args: t.Any):
         self.args: t.Dict[str, t.Any] = args
         self.parent: t.Optional[Expression] = None
         self.arg_key: t.Optional[str] = None
         self.comments: t.Optional[t.List[str]] = None
         self._type: t.Optional[DataType] = None
-        self._meta: t.Optional[t.Dict[str, t.Any]] = None
+        self.meta: t.Optional[t.Dict[str, t.Any]] = {}
```

Comment:
This should be `None` by default, and only set to a dictionary lazily if the user tries to access it. You need a `@property` that first checks if the internal field is `None`, in order to initialize it properly and return the dict.